infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/impl/AbstractSszPrimitiveSchema.java

```diff
@@ -37,10 +38,25 @@
  *
  * @param <SszDataT> Class of the basic view of this type
  */
-public abstract class AbstractSszPrimitiveSchema<
-        DataT, SszDataT extends SszPrimitive<DataT, SszDataT>>
+public abstract class AbstractSszPrimitiveSchema<DataT, SszDataT extends SszPrimitive<DataT>>
     implements SszPrimitiveSchema<DataT, SszDataT> {
 
+  protected static Bytes updateExtending(Bytes origBytes, int origOff, Bytes newBytes) {
```

Comment:
im surprised you got away with not using `final`, but we should add...

Reply:
I hope I fixed this everywhere. 
I believe 'final' was optional when I was doing my last PR )))